aep/general/0135/aep.md.j2

```diff
@@ -0,0 +1,314 @@
+}
+```
+
+- The response type **must** be set to the appropriate return type if the RPC
+  was not long-running: `google.protobuf.Empty` for most Delete RPCs, or the
+  resource itself for soft delete ([AEP-164][aep-164]).
+- Both the `response_type` and `metadata_type` fields **must** be specified
+  (even if they are `google.protobuf.Empty`).
+
+{% endtabs %}
+
+### Cascading delete
+
+Sometimes, it may be necessary for users to be able to delete a resource as
+well as all applicable child resources. However, since deletion is usually
+permanent, it is also important that users not do so accidentally, as
+reconstructing wiped-out child resources may be quite difficult.
+
+If an API allows deletion of a resource that may have child resources, the API
+**should** provide a `bool force` field on the request, which the user sets to
+explicitly opt in to a cascading delete.
+
+{% tab oas -%}
+
+{% sample 'cascading_delete.oas.yaml', 'paths' %}
+
+The API **must** fail with a `409 Conflict` error if the `force` field
+is `false` (or unset) and child resources are present.
+
+{% tab proto -%}
+
+```proto
+message DeletePublisherRequest {
+  // The name of the publisher to delete.
+  // Format: publishers/{publisher}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/Publisher"
+    }];
+
+  // If set to true, any books from this publisher will also be deleted.
+  // (Otherwise, the request will only work if the publisher has no books.)
+  bool force = 2;
+}
+```
+
+The API **must** fail with a `FAILED_PRECONDITION` error if the `force` field
+is `false` (or unset) and child resources are present.
+
+{% endtabs %}
+
+### Protected delete
+
+Sometimes, it may be necessary for users to ensure that no changes have been
+made to a resource that is being deleted. If a resource provides an [etag][],
+the delete request **may** accept the etag (as either required or optional):
+
+{% tab oas -%}
+
+{% sample 'protected_delete.oas.yaml', 'paths' %}
+
+If the etag is provided and does not match the server-computed etag, the
+request **must** fail with `412 Precondition Failed`.
+
+{% tab proto -%}
+
+```proto
+message DeleteBookRequest {
+  // The name of the book to delete.
+  // Format: publishers/{publisher}/books/{book}
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/Book"
+    }];
+
+  // Optional. The etag of the book.
+  // If this is provided, it must match the server's etag.
+  string etag = 2;
+}
+```
+
+If the etag is provided and does not match the server-computed etag, the
+request **must** fail with a `ABORTED` error code.
+
+{% endtabs %}
+
+**Note:** Declarative-friendly resources ([AEP-128][aep-128]) **must** provide the `etag`
+field for Delete requests.
+
+### Delete if existing
+
+If the service uses client-assigned resource names, `Delete` methods **may**
```

Comment:
I think this should be removed.

I've never understood the rationale for this field, which would basically cause the API to behave differently in a way the client could handle if they so wished. i.e. if the server returns a 404, then just ignore it in the client.

Reply:
I'm fine with removing this -- we should simplify wherever possible.